### PR TITLE
Draft: Suggestions in Zen mode

### PR DESCRIPTION
**Pull request by juliancoffee**
## Proposed changes

Solves #671, kind of.
At the time of writing, it is functional, but probably need to move most events to javascript to avoid resending Zen each time.
Also, I went via always show suggestions route, but I think that probably may be configured.

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [ ] I have squashed my commits into logic units.
- [ ] I have described the changes in the commit messages.

## Summary by CodeRabbit

- **New Features**
	- Introduced permission checks for user interactions on suggestions in the translation unit interface.
	- Added a form for users to submit suggestions, enhancing user engagement.
- **Bug Fixes**
	- Adjusted the layout of the voting buttons for improved visual positioning.

### CODE REVIEW

File: weblate/templates/snippets/suggestions.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
         {% else %}
         {% perm 'suggestion.delete' suggestion as user_can_delete_suggestion %}
         {% if user_can_vote_suggestion or user_can_accept_suggestion or user_can_delete_suggestion %}
-        <div class="btn-float pull-right flip">
+        <div class="btn-float pull-right flip" style="top: -20px">
```
Comment (nijel) [inline weblate/templates/snippets/suggestions.html:28]: Please avoid inline styles.

File: weblate/trans/views/edit.py [hunk 2]
```diff
@@ -893,6 +893,32 @@ def zen(request, path):
...
+            or "delete" in request.POST
+            or "spam" in request.POST
+            or "upvote" in request.POST
+            or "downvote" in request.POST
+        )
+    ):
+        # Handle accepting/deleting suggestions
+        _obj, unit_set, _context = parse_path_units(
+            request, path, (Translation, ProjectLanguage, CategoryLanguage)
+        )
```
Comment (nijel) [inline weblate/trans/views/edit.py:912]: This is already done at top of this function.

File: weblate/trans/views/edit.py [hunk 3]
```diff
@@ -893,6 +893,32 @@ def zen(request, path):
...
+        and "merge" not in request.POST
+        and (
+            "accept" in request.POST
+            or "accept_edit" in request.POST
+            or "accept_approve" in request.POST
+            or "delete" in request.POST
+            or "spam" in request.POST
+            or "upvote" in request.POST
+            or "downvote" in request.POST
+        )
```
Comment (nijel) [inline weblate/trans/views/edit.py:907]: This is essentially a copy of code from `translate` function, it should be shared.

File: weblate/trans/views/edit.py [hunk 4]
```diff
@@ -893,6 +893,32 @@ def zen(request, path):
...
+
+        checksum_form = ChecksumForm(unit_set, request.POST)
+        if not checksum_form.is_valid():
+            show_form_errors(request, checksum_form)
+            return HttpResponseBadRequest("Invalid checksum")
+
+        unit = checksum_form.cleaned_data["unit"]
+        # We are ignoring any redirects and responses here
+        _response = handle_suggestions(request, unit, "", "")
```
Comment (nijel) [inline weblate/trans/views/edit.py:921]: If the return value is not needed, there is no need to store it in a variable.

**Comment by juliancoffee** [comment]
Thanks for the review! I want to continue working on it, just got interrupted somewhat.

File: weblate/templates/snippets/suggestions.html (hunk 1 above)
Comment (juliancoffee) [reply weblate/templates/snippets/suggestions.html:28]: yeah, I don't like it too, I wonder what should I do instead? Changing the original class doesn't sound like a good idea either, I guess I could create new one?

File: weblate/trans/views/edit.py (hunk 3 above)
Comment (juliancoffee) [reply weblate/trans/views/edit.py:907]: Can I just put such a function in the same file, or it should go to some utils.py file?

**Comment by juliancoffee** [comment]
meow, bot

**Pull request closed without merging**